n2y/utils.py

```diff
@@ -93,6 +94,51 @@ def pandoc_write_or_log_errors(pandoc_ast, format, options):
             raise
 
 
+def yaml_to_meta_value(data):
+    """
+    Convert a Python object to a Pandoc metadata values, following the approach used by
+    `yamlToMetaValue` here:
+
+    https://github.com/jgm/pandoc/blob/main/src/Text/Pandoc/Readers/Metadata.hs.
+
+    Note that all scalars end up as strings.
+    """
+    if isinstance(data, str):
+        return MetaString(data)
+    elif isinstance(data, bool):
+        return MetaBool(data)
+    elif data is None:
+        return MetaString("")
+    elif isinstance(data, numbers.Number):
+        return MetaString(str(data))
+    elif isinstance(data, list):
+        return MetaList([yaml_to_meta_value(item) for item in data])
+    elif isinstance(data, dict):
+        return MetaMap({key: yaml_to_meta_value(value) for key, value in data.items()})
```

Comment:
`if`-`else` branching is non-exhaustive. Should a terminal `else` clause be added to at least log a warning on top of the current behavior of implicitly returning `None`?

Reply:
Good Idea, I think so.